### ISSUE TITLE
redesign or revert web UI for mobile/small screens 

### ISSUE THREAD

**Issue by ooozerooo** [mastodon/mastodon#19571]
### Pitch

I know a lot of efforts have been devoted to changing the web UI layout, and i have seen #19263 about the reason for changing it, but to be honest, how mastodon looks on mobile phones now is really really weird. I dont hate changes, but what the new design is doing is cutting the screen horizontally once and then vertically once, and this ends up giving users a screen that is 15% smaller (i've calculated it) to scroll on and to look at stuff. How much time do people spend at clicking on buttons and how much time do people spend at looking at timelines? Obviously people dont need to see a full list of buttons all the time especially when it is at the expense of the amount and size of content you can look at on your screen.

Old | New
-- | --
![IMG_6641 copy](https://user-images.githubusercontent.com/106132975/198928204-4bc0ee60-f306-4ab2-b2a0-8975164267d4.png) | ![IMG_6640 copy](https://user-images.githubusercontent.com/106132975/198928164-e7150510-5697-4797-819b-431cf7800684.png)

### Motivation

I've also looked at the claimed impetus for the changes in #19487 (that thread isnt mainly on mobile UI design so i opened a new one)
>  Making better use of space on mobile devices

As i mentioned and calculated before, this obviously makes the effective space smaller. Every time i open mastodon.online/social on my phone i feel like scrolling on an iPhone mini emulator at the bottom left of my iPhone screen...

> making the look of the web app more consistent with our professionally designed native apps

I dont think the professionally designed apps have a vertical AND a horizontal crop. In fact I haven't seen any professionally designed mobile web UI doing such thing. The apps look fine to me but the mobile web UI layout looks nothing like the apps.

> cleaning up the mess of stylesheets that dealt with two different layouts across various responsive screen sizes.

I guess you do need to have different designs for different screens, because the screen of your laptop/PC has a landscape orientation and your mobile phone has a portrait one...It just makes no sense to use a same layout for everything, as screens do not only vary in sizes, they vary in orientations as well.

Hope people who spent a lot of time changing the mobile web UI layout wouldnt feel like their work has not been paid respect to, but i do think we could have a much much better web UI layout for mobile devices.

Thanks.

**Comment by Gargron** [comment]
This change has really generated a lot of complaints. Thing is, we can't go back to the old layout because of multiple new requirements (the navigation has to work for both logged in and logged out users, we do want to display the logo prominently). Maybe the sidebar could slide out through a toggle at the very top, and not be visible by default.

**Comment by AkazaRenn** [comment]
+1 for it. The new UI looks terrible. The top bar we had before for navigation had to be pushed aside but literally zero space has been saved - for the "branding" and a huge post button that doesn't do anything more than before. If you want to make logged out users able to see posts, just give them top navigation bars, too.

**Comment by AkazaRenn** [comment]
https://greasyfork.org/en/scripts/454048-no-more-sidebar-in-mastodon-4-0

Temporary solution, use at your own risk.

**Comment by ooozerooo** [comment]
> This change has really generated a lot of complaints. Thing is, we can't go back to the old layout because of multiple new requirements (the navigation has to work for both logged in and logged out users, we do want to display the logo prominently). Maybe the sidebar could slide out through a toggle at the very top, and not be visible by default.

Thank you for your reply and saying that the navigation bar could be changed in the future. 

I also wanna add that the new mobile Web UI design isn't new-user-friendly as well. If I'm not mistaken, you would like to display the logo prominently and to show more content to logged out users for branding purpose and to attract new users (which might seem unnecessary for people who are running their own instance with their own money though), but it is really not a good idea to give new users more than 10 buttons (or say, icons) all at once without describing what they are for. 

They have no idea about what the buttons stand for until they click on it, which can be a very terrible experience for people who are new here.  It might not be as intuitive as you think. Why would a new user know @ means direct messages when others apps they've used before use an envelope for DMs. Why would a new user know three little men together means 'local timeline', they might even need some time to understand what a local timeline is. If there are only 5 (as in the old layout), it wouldn't be a big problem, but giving new users more than 10 icons as a navigation bar will not do a great job for navigating them, especially considering many new users already find mastodon difficult to understand. People shouldn't have to guess and click with very little or no clue what they are going to see. I'm sure they will learn what those buttons mean eventually (like we did), but many would give up trying mastodon because they find it too confusing, and I'm sure this is not what you would like to see.

I'm saying this not because I'm just guessing it will be a problem. I run a misskey instance as well, and what seems to be a barrier for some people (even those who are familiar with fediverse) to use misskey is that, on a narrower landscape orientation screen (for example on an iPad), the navigation bar get squeezed to only a list of icons, and there are more than 10 icons there, just like what mastodon looks on mobile devices now. People keep saying to me misskey is a lot more difficult than mastodon and pleroma to understand because on some devices, when they first use misskey they see lots of buttons that they have no ideas of, and can only know what it is for after they click on every one of them. It takes time for people to understand and remember what each icon stands for, which many find troublesome and think 'im just gonna use twitter/mastodon, why bother spending a lot of time understanding a new software that I dont know if im gonna like after i understand it well'. 

**screen with enough space** 
buttons in navigation bar are much clearer to users in terms of what they are for 

<img width="1362" alt="Screen Shot 2022-11-01 at 9 14 01 pm" src="https://user-images.githubusercontent.com/106132975/199211722-906728de-163e-4be8-a5da-96384caa9ab3.png"> 

**screen with no enough space** 
you will only know what each button is for when you are very familiar with misskey, new users can find this navigation bar very confusing, and this is what mastodon is doing

<img width="1151" alt="Screen Shot 2022-11-01 at 9 13 47 pm" src="https://user-images.githubusercontent.com/106132975/199211744-9fffa611-48ee-4179-b4b5-1e720468ffc3.png">

Also, if it has to be a navigation bar on mobile devices, I think it's better to let people choose which side it's on, like what pleroma is doing for larger landscape orientation screens. It's very weird for me as a right-handed person to have the navigation bar on the right. 

I saw your post talking about how stressful and depressing handling all these things is, and I hope what I just said would not make you feel a lot worse. I run a misskey and a pleroma instance, and what keeps me using mastodon is that I think it has the best Web UI, very clear and obviously not obsessed with rounded corners. You definitely have not been given enough credit for that. What we all want is to make mastodon a better place, and it really breaks my heart to see mastodon look that awkward on my phone. I wish you all the successes you want with mastodon, and I do hope we could have a better web UI (or more choices) for mobile devices, as not only old but also new users would benefit from it.

Cheers.

**Comment by pullopen** [comment]
> we do want to display the logo prominently 

It's rather useless for a self-deployed instance, don't you think? Most of us run a non-profit instance. 


I know it would be a lot of work for UI designing/ changing. And Web-UI for mastodon had always been better than other programs, which I think is one of the reasons why it gathers most users in fediverse. 

But the new UI for 4.0.0 is both inconvienient and confusing: 

1. The navigation bar has taken too many place, especially on small screen. Now the avatars would take an exrta huge line, and the place for real content is actually occupied. It will require more effort for users to scroll down the screen for same amount of content. The new UI is almost unreadable on small screen. Also it feels really uncomfortable to read on a narrow "tape" on a narrow screen.
2. Not everyone is used to a right side bar - I mean even if you have to add a sidebar ( which is absolutely unnecessary for me, top bar is definitely better), we could at least choose which side is it on - someone would actually prefer a left one because their eyes are used to lock on the right side of the screen. - Still, for me the top/bottom bar suits way better.
3. For a new user, it would be really hard to fide where a "searching bar" is - an important function for new comers. Who would know it is hidden in a "#" panel while we are used to a 🔍 symbol?
4. When I reply to myself, my intention is to make a continious thread. And most people doso for this reason. Why would I need a line says "replies to xxx" ?
5. If a toot is in the same language with my interface language, why would I need a "translate" bottun? - Another line occupied. 

The new UI has generated mass complaint in my community. A lot of site owners I know put off upgrading because of the UI changing. So please reconsider it. Thank you!

**Comment by pullopen** [comment]
Also, one of our users reports that, when scrolling down through timeline, it's very easy to touch right sidebar by mistake. Most softwere set their sidebar on top / bottom on mobile and there's a reason for that.

**Comment by JaneValleyhasBtaken** [comment]
And this will also very inconvenient for those left-handed people. Have their considered that??

**Comment by bleubluey** [comment]
The right sidebar takes way too much space , as a right-handed user i find it kinda annoying, cuz i'd touch right sidebar by mistake repeatedly, it may be even worse for left-handed people. Also users don't always need to see 10+ icons on the screen, that only made effective space smaller, and for new users they can be intimidating. I do miss a compact layout which doesn't make me scroll all the way down to tenosynovitis, thanks.

**Comment by annebakery** [comment]
> This change has really generated a lot of complaints. Thing is, we can't go back to the old layout because of multiple new requirements (the navigation has to work for both logged in and logged out users, we do want to display the logo prominently). Maybe the sidebar could slide out through a toggle at the very top, and not be visible by default.


I am merely a 3-year Mastodon user, I am also confused about the new UI.

Like the others mentioned above, the sidebar is strange for mobile users, it cuts so much screen space. The mobile device's screen is already very small for an iOS user using an iPhone mini like me, I can't read as quickly as I used to because there's not enough space for texture and media. In my perspective, it's neither visually clear nor pleasing.

As far as I noticed, in Mastodon 4.0 there is a "translate" under every toot(post), not only in the local and federated timelines but also in your home timeline. But I (can only speak for myself) would like to follow users who mostly post using languages that I can read, so I don't think a translation can take that much of a necessity. 
For English users, the situation might vary. When I change the Interface language to English, some of the "translate" under toots(posts) would disappear, but some of them still wouldn't, I haven't found out why and where I can close this new feature. For a non-English user like me, the timeline would definitely be fulfilled with "translate". It also takes so much space that my timeline is even more limited. 
Maybe you can consider it as a feature that can be folded into the “more” button?
![Uploading QQ截图20221107143839.png…]()

Besides, the view on the computer is also a bit obscure. Just too much blank space, it's really less readable compared to before. And the mastodon icon doesn't even align with the home timeline part.
![Uploading QQ截图20221107144842.png…]()

**Comment by ooozerooo** [comment]
> When I change the Interface language to English, some of the "translate" under toots(posts) would disappear, but some of them still wouldn't, I haven't found out why and where I can close this new feature.

I think that is because some people specify a different posting language than what language they actually use for posting. If you have a look at the compose form you'll see a button to specify what language you are using for the post. It is also available in `settings > preferences > other > Posting defaults > Posting language`

<img width="300" alt="Screen Shot 2022-11-07 at 7 16 03 pm" src="https://user-images.githubusercontent.com/106132975/200263667-dda2032e-fcdb-4193-9b43-a59e02d22592.png">

If my interface language is English and I specify the post language as English, even when I actually post in Korean, there will not be a translate button (because the languages match and Mastodon assume you can understand the post without translating it). But when my interface language is English and I specify the post language as Korean, the exactly same content gets a translate button. That is because Mastodon thinks I'm a English speaker and assumes I would need translation to understand Korean.

<img width="576" alt="Screen Shot 2022-11-07 at 7 17 50 pm" src="https://user-images.githubusercontent.com/106132975/200260945-e7bc2b45-64ad-4e1b-8830-e8d4fab1a5b7.png">

Yeah and I totally agree this translation design is super weird and causes a lot of confusion. Not many people will constantly look at posts in a language that they dont understand. It'll be much better to only show a translate button when you actually expand the post but not under every post on your timeline.

**Comment by annebakery** [comment]
> Yeah and I totally agree this translation design is super weird and causes a lot of confusion. Not many people will constantly look at posts in a language that they dont understand. It'll be much better to only show a translate button when you actually expand the post but not under every post on your timeline. I dont think you can disable the translation feature if you are not the admin (you'll still need to change some codes if you are the admin though) . For a user it requires more than just adding some CSS code like `display: none` using a userstyles editor/manager browser extension to disable it, because the 'translate button' shares the same element class as the 'Read more' button (so if you make the translate button disappear, so will the 'Read more' button).

Oh!! Okay, I got it. Thanks a lot for your patient reply! (>w< ) You made this perfectly clear.
Now I understand the logic behind, though I still don't want it to stay this way...

**Comment by pullopen** [comment]
I'd like to add one thing. A toot's detailed page is extremely more unreadable on a narrow screen than the timeline with this sidebar, as the font is larger and distance between lines is wider. The sidebar has brought more pain than its convenience.

**Comment by asuh** [comment]
This is what we need on all viewport sizes:

> <table role="table">
<thead>
<tr>
<th><strong>screen with enough space</strong></th>
<th><strong>screen with no enough space</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>buttons in the navigation bar are much clearer in terms of what they are for (and the Web UI is therefore way more new-user-friendly). You can make this happen by adding a pop/slide out menu drawer (like what misskey does for mobile Web UI)</td>
<td>you will only know what each button is for when you are very familiar with the software (new users can find this navigation bar very confusing), and this is what Mastodon looks like on mobile devices now</td>
</tr>
<tr>
<td><a target="_blank" rel="noopener noreferrer nofollow" href="https://user-images.githubusercontent.com/106132975/200188316-f01588cb-39bc-4c86-a659-01932e276120.png"><img alt="Screen Shot 2022-11-07 at 5 25 15 am" src="https://user-images.githubusercontent.com/106132975/200188316-f01588cb-39bc-4c86-a659-01932e276120.png" style="max-width: 100%;" width="248"></a></td>
<td><a target="_blank" rel="noopener noreferrer nofollow" href="https://user-images.githubusercontent.com/106132975/200188332-0b37a9a9-1f71-4ccc-a396-16af73311e51.png"><img alt="Screen Shot 2022-11-07 at 5 24 56 am" src="https://user-images.githubusercontent.com/106132975/200188332-0b37a9a9-1f71-4ccc-a396-16af73311e51.png" style="max-width: 100%;" width="258"></a></td>
</tr>
</tbody>
</table>

Tweetdeck provides a similar look on its navigation bar, and you can both collapse and expand it as well as being responsive. Combining the compose column with the settings column will provide a better user experience and one place to find everything.

**Comment by mtfront** [comment]
A permanent vertical side bar on mobile doesn't make sense to me. All the other popular social apps that involves a timeline/newsfeed don't do vertical side bar for a reason. It's counter-intuitive and less functional, taking up permanent space and interrupt the fundamental experience, which is scrolling across timeline to see content. It's making change for the sake of making change IMO.

Me and a lot of other instance owner are holding off upgrade just because of these. The new version have a lot of useful features, please don't let this be the barrier of entry or a compromise that people has to make. Please seriously consider rolling it back or at least give people an option to easily hide it without digging into settings.

**Comment by wonjar** [comment]
I'm a web user and I feel too much blank on 4.0+ version UI.

**Comment by NOVAofEukarya** [comment]
Ｗhen using mastodon on mobile phone, the sidebar on the right is really a problem, I mistouch them so many times! And they are WASTING my space, I want to see stream not sidebar!!

**Comment by 10iooi01** [comment]
Move the menu bar from right to left!!

**Comment by mfa777** [comment]
Compared to uncomforting of V4, we would like to keep V3 and not upgrading. We don't need so much new features; we only want a clean and easy interface.

**Comment by aglinlin** [comment]
The new UI is so stupid that it doesn't fit into normal people's habits, especially mobile UI: Sidebars are easy to touch; have designers considered left-handed? There is too much space in the time stream, and the sidebar also occupies a large chunk of the space that should have been used to watch the time stream.Please roll back! Or leave an entry for users to choose and use their favorite UI!

**Comment by Rearaerua** [comment]
As others have said, the new mobile web UI looks **DEADLY UGLY** and **INCONVENIENT** to use due to the navigation bar changes. Where once you could just swipe to switch between pages you wanted to view, now you have to move your finger to click on confusing buttons on the navigation bar.
But I'd like to point out another issue here: the position of the **Publish**  button has been changed from the bottom right corner to the top right corner, which obviously doesn't make sense. Who would pinch the top of the screen instead of the bottom when using the phone in a vertical position? This makes it particularly inconvenient to send a post and even when I think of something fun, I hardly want to post anything.
I understand that the development team probably want to differentiate Mastodon's UI from **Twitter**, but it's not user-friendly and I'd like to see the hover publish button return to the previous position.
v4.0.1
![Screenshot_20221116_162120](https://user-images.githubusercontent.com/118334617/202144430-73c4cfd7-7f58-49b8-8948-cc6d5dc80380.jpg)
v3.5.3
![Screenshot_20221116_170158_com microsoft emmx](https://user-images.githubusercontent.com/118334617/202145445-4fab5289-274e-4b81-b96d-4bb9317cc807.jpg)
We're not totally uncomfortable with the new UI, but as mentioned in the comments above, users need to have a **choice**, not just use whatever the developers give us. I didn't see any questionnaire on timeline about whether we wanted the new UI or not until it was put into use. This is awful.
Misskey designed a different UI that can be switched, a feature that makes it very easy for users to use different devices. If we have to develop the new UI, this is exactly what Mastdon should learn.
![InkedScreenshot_20221116_162056_LI](https://user-images.githubusercontent.com/118334617/202151848-ab921237-ce83-4e2b-aed7-b6560b8d6a0a.jpg)
![Screenshot_20221116_161028_com microsoft emmx](https://user-images.githubusercontent.com/118334617/202152047-cfbf1fa4-dcbc-413d-bfdc-cf1ae83c331c.jpg)
If Mastodon wants the new UI to be friendly for **one-handed** operation on a smartphone, then the most suitable design would be to put the navigation bar at the **bottom** instead of on the sides, as Misskey and Firefox have done.
Misskey
![Screenshot_20221116_161011_com microsoft emmx](https://user-images.githubusercontent.com/118334617/202152880-a530c8ae-f724-422e-8d15-9ab8a2e27785.jpg)
Firefox
![1668593194953](https://user-images.githubusercontent.com/118334617/202152732-8a80f9ee-0797-47b0-8f42-0e2022a63377.JPEG)
We hope our suggestions will be taken on board by the development team, and that they will not only work for us old users, but also for mastodon to attract more new users.

**Comment by Cassolotl** [comment]
With the new design, when I have follow requests the bottom buttons of the menu bar are cut off at the bottom of the screen, and the menu bar isn't scrollable. So it's impossible to get to all the buttons in some situations.

<img src="https://user-images.githubusercontent.com/6607512/203010856-f0ebab54-bd50-4c75-b424-9fe2aafb2819.jpg" width=400></img>

**Comment by emshupisim** [comment]
another example of why the side navigation bar is a bad design and is extremely anti-user:

Also it is so easy to mistouch the preference icon when typing. And once you are at the preference page, all text you were precisely typing are just gone. there's no way to get them back. Having a top motivation bar is just so much better for small screens!!

**Comment by nikodunk** [comment]
> This change has really generated a lot of complaints. Thing is, we can't go back to the old layout because of multiple new requirements (the navigation has to work for both logged in and logged out users, we do want to display the logo prominently). Maybe the sidebar could slide out through a toggle at the very top, and not be visible by default.

If I made a Pull Request that did something like this on mobile-web, would you be interested in merging?

**Comment by honhatduy** [comment]
@renchap It's been three years, have we improved on this? As someone who used to love Mastodon's mobile-friendly web interface, I'm sad about this redesign.

**Comment by renchap** [comment]
We are planning for a big rework of the Mastodon web UI. I can not provide any timeframes for this, but we do not plan to change the mobile interface before this more general rework.

**Comment by honhatduy** [comment]
> We are planning for a big rework of the Mastodon web UI. I can not provide any timeframes for this, but we do not plan to change the mobile interface before this more general rework.

Great to hear. Hope you guys hit this milestone soon ;)
